### PR TITLE
feat: update openapi spec and server to include publish and setreplica

### PR DESCRIPTION
**Pull request by tiagolobocastro**
feat: update openapi spec and autogenerated code
Explicitly specify min/max for some of the integer types we use.

feat: update openapispec and model to include publish and setreplica
Add new openapi operations for creating a volume target (this means we create a nexus)
and new operation to change a volume's replica count.

feat: update rest server to include setreplica and publish
Refactor publish to return the volume.

### CODE REVIEW

File: common/src/mbus_api/message_bus/v0.rs [hunk 1]
```diff
@@ -228,6 +229,24 @@ pub trait MessageBusTrait: Sized {
         Ok(())
     }
 
+    /// set volume replica count
```
Comment (paulyoong) [inline common/src/mbus_api/message_bus/v0.rs:232]: Incorrect comment - this is the publish call

File: control-plane/rest/tests/v0_test.rs [hunk 2]
```diff
@@ -335,6 +363,32 @@ async fn client_test(mayastor: &NodeId, test: &ComposeTest, auth: &bool) {
             .unwrap()
     );
 
+    let volume = client
+        .volumes_api()
+        .put_volume_target(&volume.uuid.to_string(), None, None)
```
Comment (paulyoong) [inline control-plane/rest/tests/v0_test.rs:368]: Should we also test the case where the node is specified i.e. assert that it is published on the specified node?

**Review by paulyoong — APPROVED** [review]

File: common/src/mbus_api/message_bus/v0.rs (hunk 1 above)
Comment (tiagolobocastro) [reply common/src/mbus_api/message_bus/v0.rs:232]: dohh

File: control-plane/rest/tests/v0_test.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/rest/tests/v0_test.rs:368]: well we didn't specify a node (None) - one is auto-selected

Comment (paulyoong) [reply control-plane/rest/tests/v0_test.rs:368]: But should we also check the behaviour when one is explicitly specified?

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**